onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.h

```diff
@@ -239,12 +239,15 @@ class BaseSelector : public NodeSelector {
 
   // We std::move SelectorActionRegistry into the SelectorActionTransformer so this class needs to have a move ctor
   BaseSelector(BaseSelector&& rhs) noexcept
-      : node_group_selector_{std::move(rhs.node_group_selector_)} {
+      : node_group_selector_{std::move(rhs.node_group_selector_)},
+        compatible_providers_{std::move(rhs.compatible_providers_)} {
   }
 
  protected:
-  BaseSelector(std::unique_ptr<NodeGroupSelector> node_group_selector)
-      : node_group_selector_{std::move(node_group_selector)} {}
+  BaseSelector(std::unique_ptr<NodeGroupSelector> node_group_selector, std::vector<const char*> compatible_providers = {})
+      : node_group_selector_{std::move(node_group_selector)} {
+    compatible_providers_.assign(compatible_providers.begin(), compatible_providers.end());
```

Comment:
You should be able to call the constructor as part of the initializer list instead of calling `assign` here. e.g.:

```
: node_group_selector_{std::move(node_group_selector)},
  compatible_providers_(compatible_providers.begin(), compatible_providers.end())
```